### ISSUE TITLE
Color palette should be consistent after subsetting (with multiple axes per figure)

### ISSUE THREAD

**Issue by giovp** [scverse/scanpy#1134]
After subsetting the anndata, cluster colors should be consistent with cluster class. 
```python
sc.pl.umap(adata_spatial, color = ["clusters"], palette=sc.pl.palettes.default_28)
```
![image](https://user-images.githubusercontent.com/25887487/77832090-c2946f80-7133-11ea-8ec1-f8011a2d4c19.png)

```python
fig, axs = plt.subplots(1,2, figsize=(8,5))
sc.pl.umap(
    adata_spatial[adata_spatial.obs.library_id == "V1_Mouse_Brain_Sagittal_Anterior", :], 
    color="clusters", 
    show = False,
    ax = axs[0])

sc.pl.umap(
    adata_spatial[adata_spatial.obs.library_id == "V1_Mouse_Brain_Sagittal_Posterior", :], 
    color="clusters", 
    show = False,
    ax = axs[1])

plt.tight_layout()
```
![image](https://user-images.githubusercontent.com/25887487/77832119-f5d6fe80-7133-11ea-95bb-9e1ed9eea1cd.png)

Maybe cluster colors should be stored as dictionary with class labels as keys? Or is there another way to achieve this that I am not familiar with ?

**Comment by ivirshup** [comment]
This is already done with the `{colname}_colors` entries in `uns`. For some reason, this isn't working when `ax` is passed and `show=False` (@fidelram @flying-sheep).

Here's a reproducer using the built-in data:

```python
import scanpy as sc
from matplotlib import pyplot as plt

pbmc = sc.datasets.pbmc3k_processed()

fig, axs = plt.subplots(1,2, figsize=(8,5)) 

sc.pl.umap(
    pbmc[pbmc.obs["louvain"].isin(['Dendritic cells', 'Megakaryocytes'])],
    color="louvain",
    show=False,
    ax=axs[0]
)
# Trying to set attribute `.uns` of view, copying.
sc.pl.umap(pbmc, color="louvain", show=False, ax=axs[1])

plt.show()
```

<img width="922" alt="image" src="https://user-images.githubusercontent.com/8238804/77877321-2d16ee00-72a1-11ea-99e7-1d7eb75d036c.png">

Interestingly, if you plot the full object before the view this works fine:

```python
fig, axs = plt.subplots(1,2, figsize=(8,5)) 

sc.pl.umap(pbmc, color="louvain", show=False, ax=axs[1])
sc.pl.umap(
    pbmc[pbmc.obs["louvain"].isin(['Dendritic cells', 'Megakaryocytes'])],
    color="louvain",
    show=False,
    ax=axs[0]
)

plt.show()
```

<img width="912" alt="image" src="https://user-images.githubusercontent.com/8238804/77877417-736c4d00-72a1-11ea-9aad-1afb98089c25.png">

**Comment by ivirshup** [comment]
Fixed by theislab/anndata#388
